src/edu/csus/ecs/pc2/clics/API202306/CLICSFileReference.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 1989-2024 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.clics.API202306;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonFilter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ser.FilterProvider;
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+
+/**
+ * CLICS File Reference
+ * Contains information about a file used in the API
+ *
+ * @author John Buck
+ *
+ */
+
+@JsonFilter("rtFilter")
+public class CLICSFileReference {
+
+    @JsonProperty
+    private String href;
+
+    @JsonProperty
+    private String filename;
+
+    @JsonProperty
+    private String hash;
+
+    @JsonProperty
+    private String mime;
+
+    @JsonProperty
+    private int width;
+
+    @JsonProperty
+    private int height;
+
+    @JsonProperty
+    private String data;
+
+    /*
+     * For Jersey deserialization
+     */
+    public CLICSFileReference() {
+
+    }
+
+    /**
+     * Fills in properties for a CLICS FILE reference.  Currently, we only support the required Href, Filename and Mime type.
```

Comment:
We should probably file a separate issue for upgrading to include actual support for MIME image types, since I suspect we're going to want that for things like school logos...

Reply:
The class sort of, kind of supports different mime types now.  However, as you point out for images, it should probably fill in the width and height properties if the mime type is some sort of image.   Someday.